packages/teams-js/src/private/externalAppAuthentication.ts

```diff
@@ -243,14 +291,14 @@ export namespace externalAppAuthentication {
    * Signals to the host to perform authentication using the given authentication parameters and then resend the request to the application specified by the app ID with the authentication result.
    * @internal
    * Limited to Microsoft-internal use
-   * @param appId ID of the application backend to which the request and authentication response should be sent
+   * @param appId ID of the application backend to which the request and authentication response should be sent. This must be a UUID
    * @param authenticateParameters Parameters for the authentication pop-up
    * @param originalRequestInfo Information about the original request that should be resent
    * @returns A promise that resolves to the IInvokeResponse from the application backend and rejects with InvokeError if the host encounters an error while authenticating or resending the request
    */
   export function authenticateAndResendRequest(
     appId: string,
```

Comment:
Should this be a UUID then?

Reply:
Same applies to all places we use `appId`

Reply:
Discussed offline. There doesn't seem to be a good type we can pull from the `uuid` package that we can use as an input type here, so I think it would be better to just leave the input type as string add UUID validation (uuid.validate) here. Small caveat: it looks like the app manifest allows capitalization in the ID while the UUID validation does not, but we can just cast toLower before we validate.